File: src/1422. Maximum Score After Splitting a String/soln.java
//Solution to 1422. Maximum Score After Splitting a String(https://leetcode.com/problems/maximum-score-after-splitting-a-string/)





class Solution {
    public int maxScore(String s) {

        int ans =0,zero=0;
        int totalOnes = 0;

        for(char c : s.toCharArray()){
            if(c=='1')
                totalOnes++;
        }

        for(int i=0;i<s.length()-1;i++){
            if(s.charAt(i)=='0'){
                ++zero;
            }
            else{
                --totalOnes;
            }
            ans = Math.max(ans, zero+totalOnes);
        }
        return ans;
        
    }
}

File: src/1422. Maximum Score After Splitting a String/soln.c
//Solution to 1422. Maximum Score After Splitting a String(https://leetcode.com/problems/maximum-score-after-splitting-a-string/)


int maxScore(char* s) {
    int ans = 0;
    int zeros = 0;
    int ones = 0;

    for(int i=0;s[i]!='\0';i++){
        if(s[i]=='1')
            ones++;
    }
    for(int i=0;s[i+1]!='\0';i++){
        if(s[i]=='0')
            zeros++;
        else
            ones--;

        int score = zeros+ones;
        if(score>ans){
            ans = score;
        }

    }
    return ans;
}





